src/libopensc/iasecc-sdo.c

```diff
@@ -143,20 +143,22 @@ iasecc_sdo_convert_acl(struct sc_card *card, struct iasecc_sdo *sdo,
 void
 iasecc_sdo_free_fields(struct sc_card *card, struct iasecc_sdo *sdo)
 {
-	if (sdo->docp.tries_remaining.value)
-		free(sdo->docp.tries_remaining.value);
-	if (sdo->docp.usage_remaining.value)
-		free(sdo->docp.usage_remaining.value);
-	if (sdo->docp.non_repudiation.value)
-		free(sdo->docp.non_repudiation.value);
-	if (sdo->docp.acls_contact.value)
-		free(sdo->docp.acls_contact.value);
-	if (sdo->docp.size.value)
-		free(sdo->docp.size.value);
-	if (sdo->docp.name.value)
-		free(sdo->docp.name.value);
-	if (sdo->docp.issuer_data.value)
-		free(sdo->docp.issuer_data.value);
+	free(sdo->docp.tries_maximum.value);
```

Comment:
In iasecc_sdo_allocate_and_parse, struct iasecc_sdo is created with calloc(). Since you're adding resets to NULL to some members, you may want to extend this to the whole structure by using memset(sdo, 0, sizeof *sdo);. This would also destroy the `magic` value, which I think is a good thing, because the structure is not valid anymore.